src/prometheus_server.py

```diff
@@ -1,19 +1,38 @@
-import json
-import urllib3
+import logging
+from requests import get, post
+from requests.exceptions import ConnectionError, ConnectTimeout
+
+logger = logging.getLogger(__name__)
 
 
 class Prometheus:
-    def __init__(self, host, port):
+    def __init__(self, host: str, port: int, api_timeout=2.0):
         """Utility to manage a Prometheus application.
         Args:
             host: host address of Prometheus application.
             port: port on which Prometheus service is exposed.
         """
-        self.host = host
-        self.port = port
-        self.http = urllib3.PoolManager()
+        self.base_url = "http://{}:{}".format(host, port)
+        self.api_timeout = api_timeout
+
+    def reload_configuration(self) -> bool:
+        """Send a POST request to to hot-reload the config.
+        This reduces down-time compared to restarting the service.
+        Returns:
+          True if reload succeeded (returned 200 OK); False otherwise.
+        """
+        url = "{}/-/reload".format(self.base_url)
+        try:
+            response = post(url, timeout=self.api_timeout)
+
+            if response.status_code == 200:
+                return True
+        except (ConnectionError, ConnectTimeout) as e:
+            logger.debug("config reload error via %s: %s", url, str(e))
+
+        return False
 
-    def build_info(self):
+    def build_info(self) -> dict:
```

Comment:
```suggestion
    def build_info(self) -> Dict[str, str]:
```

Reply:
I am fairly sure we need an additional import though